saleor/graphql/product/types/products.py

```diff
@@ -1090,37 +1148,54 @@ def resolve_image_by_id(root: ChannelContext[models.Product], _info, *, id):
         return root.node.media.filter(pk=pk).first()
 
     @staticmethod
-    def resolve_media(root: ChannelContext[models.Product], info):
-        return MediaByProductIdLoader(info.context).load(root.node.id)
+    def resolve_media(root: ChannelContext[models.Product], info, sort_by=None):
+        if sort_by is None:
+            sort_by = {
+                "field": MediaChoicesSortField.ID.value,  # type: ignore
+                "direction": "",
+            }
+        return MediaByProductIdLoaderSort(info.context, sort_by).load(root.node.id)
 
     @staticmethod
     def resolve_images(root: ChannelContext[models.Product], info):
         return ImagesByProductIdLoader(info.context).load(root.node.id)
 
     @staticmethod
-    def resolve_variants(root: ChannelContext[models.Product], info):
-        requestor = get_user_or_app_from_context(info.context)
-        has_required_permissions = has_one_of_permissions(
-            requestor, ALL_PRODUCTS_PERMISSIONS
-        )
-        if has_required_permissions and not root.channel_slug:
-            variants = ProductVariantsByProductIdLoader(info.context).load(root.node.id)
-        elif has_required_permissions and root.channel_slug:
-            variants = ProductVariantsByProductIdAndChannel(info.context).load(
-                (root.node.id, root.channel_slug)
-            )
-        else:
-            variants = AvailableProductVariantsByProductIdAndChannel(info.context).load(
-                (root.node.id, root.channel_slug)
+    def resolve_variant(root: ChannelContext[models.Product], info, id=None, sku=None):
+        if id is None and sku is None:
+            return None
+
+        keys = {}
+        if id is not None:
+            _, pk = from_global_id_or_error(id, ProductVariant)
+            keys["id"] = int(pk)
+
+        if sku is not None:
+            keys["sku"] = sku
+
+        def get_product_variant_by_id_and_sku(
+            product_variants, keys
+        ) -> Optional[ProductVariant]:
+            def condition(variant):
+                con = True
+                for key, val in keys.items():
```

Comment:
What are `key` and `value`? IMO, we should name it with something meaningful.  For example, instead of `key`, we could have `field_name` and `val` could be `value`.